csharp/SecureMemory/SecureMemory/ProtectedMemoryImpl/Libc/LibcProtectedMemoryAllocatorLP64.cs

```diff
@@ -50,13 +48,6 @@ public virtual void SetReadWriteAccess(IntPtr pointer, ulong length)
         // ************************************
         public virtual IntPtr Alloc(ulong length)
         {
-            libc.getrlimit(GetMemLockLimit(), out var rlim);
-            if (rlim.rlim_max != rlimit.UNLIMITED && rlim.rlim_max < length)
-            {
-                throw new MemoryLimitException(
-                    $"Requested MemLock length exceeds resource limit max of {rlim.rlim_max}");
-            }
-
```

Comment:
Why was this check removed?

Reply:
This check is comparing the global memory limit against the individual allocation size.  It would make sense if we tracked the number of bytes we allocated/deallocated and added that value to length and THEN compared it to the global mlock limit, but otherwise it only catches one unlikely case, where any *single* allocation exceeds the entire mlock limit.

Reply:
I agree that this check is of limited value, but in the unlikely case of the single allocation, wouldn't this message be more valuable than the dreaded `Libc call mlock failed with result -1 12` that would occur otherwise?

Reply:
I've replaced the check with tracking of actual memory locked and comparing that value to the limit.